TwitchLib.Client.Models/SubscriberBase.cs

```diff
@@ -165,7 +165,9 @@ protected SubscriberBase(IrcMessage ircMessage)
                             case "":
                                 break;
                             default:
-                                throw new ArgumentOutOfRangeException(nameof(tagValue.ToLower));
+                                // TODO: rather logging than throwing an exception
```

Comment:
Where I do agree with this, it has been discussed in length in the past that Swifty wanted them thrown. @swiftyspiffy  If that is still the case or not?

Reply:
For me that popped up while running integration tests in Debug-Mode from inside VS.

It's just an explanation, why I did and think so (in other cases/classes too; this is just an example):

- `TwitchClient.Client_OnMessage(object sender, OnMessageEventArgs e)` invokes `TwitchClient.HandleIrcMessage(IrcMessage ircMessage)`
- `TwitchClient.HandleIrcMessage(IrcMessage ircMessage)` invokes `TwitchClient.HandleUserNotice(IrcMessage ircMessage)`
- `TwitchClient.HandleUserNotice(IrcMessage ircMessage)` comes to the `case MsgIds.Subscription`
- `Subscriber subscriber = new Subscriber(ircMessage);` is called
- ctor `public Subscriber(IrcMessage ircMessage) : base(ircMessage) { }` passes ircMessage to `SubscriberBase` s ctor
- now the tag-values become evaluated within an `for`-loop
- by `throw`ing an `Exception`, everything is gone, not only the `instance` that should be created, but also the `throw`n `Exception` cause it never gets `catch`ed, at this very moment

So I think its better to have an `Instance` with a missing `value` that can be tracked within log-files or wherever than having 'nothing' but the .


While takin another closer look and rethinkin,
another approach to solve that and must-do (from my point of view),
no matter what the decision for those TODOs is,
is to add a 'master'-`try-catch`-block within `TwitchClient.Client_OnMessage(object sender, OnMessageEventArgs e)` around `TwitchClient.HandleIrcMessage(IrcMessage ircMessage)`
that `catch`es any exception that could be `throw`n

So, what about that?